src/execution_plan/execution_plan_build/execution_plan_construct.c

```diff
@@ -91,6 +95,10 @@ void ExecutionPlan_RePositionFilterOp(ExecutionPlan *plan, OpBase *lower_bound,
 			ExecutionPlan_RemoveOp(plan, (OpBase *)filter);
 			ExecutionPlan_PushBelow(op, (OpBase *)filter);
 		}
+	} else if(op == NULL) {
```

Comment:
is it possible to reach the previous block `if(filter->childCount > 0) {` when `op == NULL` ?

Reply:
I don't think it should be, the previous block only triggers when we're placing filters in the ExecutionPlan optimization phase (at which time we will always have a plan and root op).